src/evmTest/scala/io/iohk/ethereum/vm/ContractCallingItselfSpec.scala

```diff
@@ -0,0 +1,23 @@
+package io.iohk.ethereum.vm
+
+import io.iohk.ethereum.vm.utils.EvmTestEnv
+import org.scalatest.{FreeSpec, Matchers}
+
+// scalastyle:off magic.number
+class ContractCallingItselfSpec extends FreeSpec with Matchers {
+
+  "EVM running ContractCallingItself contract" - {
+
+    "should handle a call to itself" in new EvmTestEnv {
+      val (_, contract) = deployContract("ContractCallingItself")
+
+      contract.getSomeVar().call().returnData shouldBe UInt256(10).bytes
+
+      val result = contract.doubleSomeVar().call()
```

Comment:
Minor detail: As the contract is called `ContractCallingItself` shouldn't this test call here `callSelf` to test it?